phpunit/experimental/rest-api-comment-permissions-test.php

```diff
@@ -0,0 +1,211 @@
+		$draft_id = $this->factory->post->create(
+			array(
+				'post_status' => 'draft',
+			)
+		);
+		$params   = array(
+			'post'         => $draft_id,
+			'author_name'  => 'Ishmael',
+			'author_email' => 'herman-melville@earthlink.net',
+			'author_url'   => 'https://en.wikipedia.org/wiki/Herman_Melville',
+			'content'      => 'Call me Ishmael.',
+			'author'       => self::$user_ids['editor'],
+			'comment_type' => 'block_comment',
+		);
+
+		$request = new WP_REST_Request( 'POST', '/wp/v2/comments' );
+		$request->add_header( 'Content-Type', 'application/json' );
+		$request->set_body( wp_json_encode( $params ) );
+
+		$response    = rest_get_server()->dispatch( $request );
+		$data        = $response->get_data();
+		$new_comment = get_comment( $data['id'] );
+		$this->assertSame( 'Call me Ishmael.', $new_comment->comment_content );
+	}
+
+	/**
+	 * Test that for each user role, the permissions are correct when accessing comments.
+	 *
+	 * @param string $role The user role to test.
+	 * @param string $post_author_role The role of the post author.
+	 * @param bool $can_read The expected permission result.
+	 *
+	 * @dataProvider data_get_block_comment_permissions_data_provider
+	 */
+	public function test_get_block_comment_permissions_edit_context( $role, $post_author_role, $can_read ) {
+		wp_set_current_user( self::$user_ids[ $role ] );
+		$post_id = $this->create_test_post_with_block_comment( self::$user_ids[ $post_author_role ] );
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/comments' );
+		$request->set_param( 'post', $post_id );
+		$request->set_param( 'type', 'block_comment' );
+		$request->set_param( 'status', 'all' );
+		$request->set_param( 'per_page', 100 );
+		$request->set_param( 'context', 'view' );
+		$response = rest_get_server()->dispatch( $request );
+
+		if ( $can_read ) {
+			$comments = $response->get_data();
+			$this->assertEquals( self::$num_comments, count( $comments ) );
+
+		} else {
+			$this->assertErrorResponse( 'rest_forbidden_context', $response, 403 );
+		}
+
+		wp_delete_post( $post_id, true );
+	}
+
+	public function data_get_block_comment_permissions_data_provider() {
+		return array(
+			'Administrator can see block comments on other posts' => array( 'administrator', 'author', true ),
+			'Editor can see block comments on other posts' => array( 'editor', 'contributor', true ),
+			'Author cannot see block comments on other posts' => array( 'author', 'editor', false ),
+			'Contributor cannot see block comments on other posts' => array( 'contributor', 'author', false ),
+			'Subscriber cannot see block comments'         => array( 'subscriber', 'contributor', false ),
+			'Author can see block comments on own post'    => array( 'author', 'author', true ),
+			// Reason: It only returns partial data for contributors, only unit tests, not sure why.
+			// 'Contributor can see block comments on own post' => array( 'contributor', 'contributor', true ),
+		);
+	}
```

Comment:
@adamsilverstein, based this on your data provider example. The last commented-out case isn't working for me via unit tests, but I can't reproduce it manually.

The response returns 3-5 comments, instead of 10.

Reply:
Figured it out 🤦 Contributors can't edit the published posts.